Sources/Xcodeproj/pbxproj().swift

```diff
@@ -530,6 +530,29 @@ func xcodeProject(
             }
         }
     }
+    
+    // Create an aggregate target for every product for which there isn't already
+    // a target with the same name.
+    let targetNames: Set<String> = Set(modulesToTargets.values.map{ $0.name })
+    for product in graph.products {
+        // Go on to next product if we already have a target with the same name.
+        if targetNames.contains(product.name) { continue }
+        // Otherwise, create an aggreate target.
+        let target = project.addTarget(productType: nil, name: product.name)
```

Comment:
It's not entirely clear what the right thing to do is here.  Making them aggregate targets is a first step, so they can at least be built.  My concern with making them be the product type that is specified in the product declaration is that then we'll have two Xcode targets for building some of the Swift package targets.

But I agree, it doesn't feel totally right.  Really, what should happen, is that the Xcode project generation should be reevaluated now that we have product types.  I could see an argument for saying that teach product should generate an Xcode target of the appropriate type, and we should not generate any targets at all for package modules other than those.

Was the Xcode project generation logic updated as part of the product proposal implementation?